src/System.Management.Automation/engine/hostifaces/HostUtilities.cs

```diff
@@ -811,6 +832,192 @@ public static Collection<PSObject> InvokeOnRunspace(PSCommand command, Runspace
+        /// <param name="text">The text to be rendered.</param>
+        /// <param name="style">The style to be used.</param>
+        /// <param name="ansiReset">The ANSI code to reset.</param>
+        /// <param name="indent">The number of spaces for indentation.</param>
+        /// <param name="startOnNewLine">Indicates whether to start writing from a new line.</param>
+        internal static void RenderText(StringBuilder output, string text, string style, string ansiReset, int indent, bool startOnNewLine)
+        {
+            if (text is null)
+            {
+                return;
+            }
+
+            if (startOnNewLine)
+            {
+                // Start writing the text on the next line.
+                output.AppendLine();
+            }
+
+            // Apply the style.
+            output.Append(style);
+
+            int count = 0;
+            var trimChars = "\r\n".AsSpan();
+            var span = text.AsSpan().Trim(trimChars);
+
+            // This loop renders the text with minimal allocation.
+            while (true)
+            {
+                int index = span.IndexOf('\n');
+                var line = index is -1 ? span : span.Slice(0, index);
+
+                if (startOnNewLine || count > 0)
+                {
+                    output.Append(' ', indent);
+                }
+
+                output.Append(line.Trim('\r')).AppendLine();
+
+                // Break out the loop if we are done with the last line.
+                if (index is -1)
+                {
+                    break;
+                }
+
+                // Point to the rest of feedback text.
+                span = span.Slice(index + 1);
+                count++;
+            }
+
+            output.Append(ansiReset);
+        }
+
+        /// <summary>
+        /// Helper function to render feedback actions.
+        /// </summary>
+        /// <param name="output">The output string builder to write to.</param>
+        /// <param name="item">The feedback item to be rendered.</param>
+        /// <param name="textStyle">The style used for feedback messages.</param>
+        /// <param name="actionStyle">The style used for feedback actions.</param>
+        /// <param name="ansiReset">The ANSI code to reset.</param>
+        internal static void RenderActions(StringBuilder output, FeedbackItem item, string textStyle, string actionStyle, string ansiReset)
+        {
+            if (item.RecommendedActions is null || item.RecommendedActions.Count is 0)
+            {
+                return;
+            }
+
+            List<string> actions = item.RecommendedActions;
+            if (item.Layout is FeedbackItem.DisplayLayout.Landscape)
+            {
+                // Add 4-space indentation and write the indicator.
+                output.Append(' ', 4)
+                    .Append(textStyle).Append(s_actionIndicator).Append(ansiReset)
+                    .Append(' ');
```

Comment:
The same for interpolated string handler overload. Below too.

Reply:
Updated.